refactor: next/router hooks with next/navigation hooks

grzpab opened the pull request:
## What does this PR do?

Moves the project from Next.js 13.2 to 13.4. 
Replaces all usages of hooks from `next/router` to `next/navigation`

## Type of change
- Chore (refactoring code, technical debt, workflow improvements)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How should this be tested?
Full application check, including visual regression checks.

## Checklist
- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works

grzpab committed:
[Intuita] codemod: "replace next router"  (107 files changed, estimated: ~64% of changes in the PR)

grzpab committed:
Manual fixes for tricky cases intertwined with business logic (61 files changed, estimated: ~36% of changes in the PR)

grzpab committed:
[Intuita] bump the nextjs version to ^13.4.1

grzpab committed:
[Intuita] Keep the PR fresh: Revert "Unstage enums file (#8836)"

This reverts commit c873b839654d01e769249f4f90dc55cb42a584dc.

grzpab committed:
[Intuita] keep the PR fresh: post-rebase fix

zomars reviewed (changes requested):
Blocking to undo unrelated formatting changes to this PR. Quite important since this PR touches a lot of files.

Also can you post what does the Next.js docs say about this deprecation?

zomars commented on apps/web/components/apps/layouts/AppsLayout.tsx:
```diff
@@ -13,14 +13,11 @@ type AppsLayoutProps = {
   actions?: (className?: string) => JSX.Element;
   emptyStore?: boolean;
 } & Omit<ComponentProps<typeof Shell>, "actions">;
-
```
Same here. An the rest of these kind of diffs. Our rationale is that we cannot be adapting stylistic preferences for each contributor so we tend to avoid formatting changes were they're not required.

zomars commented on apps/web/components/NavTabs.tsx:
```diff
@@ -95,5 +97,4 @@ const NavTabs: FC<NavTabProps> = ({ tabs, linkProps, ...props }) => {
     </>
   );
 };
-
```
>  In the meantime, do you think using https://eslint.org/docs/latest/rules/padding-line-between-statements on top of prettier could help?

This could help, but not for this PR. I think this will cause even more unnecessary diffs since this isn't enforced consistently through our codebase.

emrysal committed:
Tweak/8843 fix merge conflicts next navigation pr (#1)

* fix: overflow (#8858)

* Fix/ Routing Form Routes (#8827)

* Reuse split routes

* Add cache

* Update formMutation.handler.ts

---------

Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>

* Feat/team invite flow (#8804)

* completeOnboarding drives invite flow changes

* Username prefil - force getting started when acc doesnt exist

* Change subtitle text and unique error on multiple invites

* User exists and no password set - generate signup link still

* Fix callback param

* Update packages/trpc/server/routers/viewer/teams/inviteMember.handler.ts

* Fix spacing

* Add password hints

* Fix translations

---------

Co-authored-by: Bailey Pumfleet <bailey@pumfleet.co.uk>

* Adding icon.svg and renaming config file (#8859)

* chore: removed duplicate value in className @ Tooltip.tsx (#8865)

* New Crowdin translations by Github Action

* Pull latest main in every submodule (#8537)

* Update to latest main so that yarn.lock is always correct

* Update git-setup.sh

* a11y: Exit the lexical Editor and button a11y on tab index (#8139)

* a11y: Exit the lexical Editor and button a11y on tab index

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

* chore: Updated lexical to v0.9.0

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

* chore: Added ErrorBoundary for Lexical RichTextPlugin

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

* chore: yarn install changes

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>

* fix: merge conflicts

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>

* Update yarn.lock

---------

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>
Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: zomars <zomars@me.com>

* Refactor/readme.md (#8723)

* Update README.md 

Removed 
- Heroku DB setup section (since it's no longer free). 
Added
- Section for setting up local DB.
- Some other minor details.

* Minor grammer fixes

---------

Co-authored-by: vaibhav bisht <vaibhav@aus.co.in>
Co-authored-by: Keith Williams <keithwillcode@gmail.com>

* cancel and schedule reminders when workflow activated/deactivated on event type (#8757)

* add switches to advanced even type settings

* only show switches when workflow is enabled

* check in event type update handler if user is allowed to disable standard emails

* don't send emails if disabled

* remove  reminders when workflow is deactivated on event type

* make sure emails can't be disabled if no workflow exist

* send workflow emails to all attendees

* code clean up

* add translations

* always send to all attendees when scheduling reminders

* schedule reminders if event type gets activated on workflow

* schedule reminders (commented code)

* change text

* schedule reminders for all attendees not just first one

---------

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>
Co-authored-by: zomars <zomars@me.com>

* Fix: #8528 Moved the data checks to superRefine (#8749)

* Fix: #8528 Moved the data checks to superRefine

* Resolved lint isseu

* Updated to parse from safeParse

* added new line

---------

Co-authored-by: Omar López <zomars@me.com>

* New Crowdin translations by Github Action

* Add changesets/cli to release embed (#8126)

* Add dep versions

* Release new version for embeds

* Add prepublish

* Use prepare script which works with both npm and yarn

* Add changesets correctly

* Add embed publishing command

---------

Co-authored-by: zomars <zomars@me.com>

* Fix hidden label breaking layout (#8867)

* Star not centered issue #8519 (#8872)

* modifications for the issues #8159 are done

* modifications for the issue #8519 are made

* only border top for bottom nav (#8882)

* Feat: select all days in schedule (#8796)

* feat: add select all

* fix: fix deleted days

* chore: add i18n select all

* chore: check all days when select all

* chore: use calcom checkbox

---------

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>

* Fixed dark mode bugs in the availability section (#8630)

* Fixed dark mode bugs in the availability section

* Reverted the tooltip change for delete button

* Updated tooltip of delete button conditionally based on isLastSchedule flag

* fix: disabled button state

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>

---------

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>

* Add eventTypeSelected event (#8894)

* fix: strip markdown from meta description tag in users page (#8769)

* fix: remove markdown validation

* feat: use markdown stripped bio

* fix: use remove markdown

* fix: type errors

---------

Co-authored-by: alannnc <alannnc@gmail.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

* [fix] : select options marginal space (#8892)

Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>

* Gitpod setup And Bio Detail Overflow Bug Fixed (#8824)

* feat: Add Gitpod setup and update README.md

* fix: Fix overflow issue in bio details with word-wrap property

* minor changes in gitpod script

* updated gitpod url for cal

* gitpod url fixed in markdown

* fix: lint errors

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>

* added vscode extensions and  enabled prebuilds to reduce waiting time

---------

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>

* Enabling workflows for managed events (#8446)

* Enabling workflows for managed events

* Disabling fields for workflow when readonly

* Disabling event type workflows if readonly

* Installing auth app

* Reverting yarn.lock

* Again reverting yarn

* Showing user workflows in managed event type

* Reusing existing code to show all workflows

* Further extending workflow support

* Added unit test

* Fixing workflow assignment

* Fixing locked workflows for children MET

* Update packages/features/ee/workflows/components/EventWorkflowsTab.tsx

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>

* Personal workflows fix, matching count numbers

* Fixing menu items for managed event types

* Fixing type checks

* Fixing empty activeOn select

* Fixing unit test

* handling active workflows for MET

* Fixing MET slug nit

* Embed option in action restored for non-MET

* Correcting mobile view when hidden

* More adjustments

* fix sms reminder field for children

* remove console.log

---------

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
Co-authored-by: CarinaWolli <wollencarina@gmail.com>

* New Crowdin translations by Github Action

* CAL-1557: Improve timeformat selector alignment in new booker (#8898)

* CAL-1664: Improve flash of toggle group when rerendering (#8900)

* CAL-1664: Improve flash of toggle group when rerendering, because active element moves into position.

* Added back newline

* CAL-1550: Align form submit button of new booker to bottom (#8896)

Co-authored-by: Alex van Andel <me@alexvanandel.com>

* v2.9.0

* New Crowdin translations by Github Action

* fix: add exp in /meeting-tokens (#8869)

* fix: add exp in /meeting-tokens

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>

* hijacking this PR to add a NIT change lol

---------

Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Peer Richelsen <peer@cal.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

* Fix yarn.lock (missing lockfile commit in #8769)

---------

Signed-off-by: GurkiranSingh <gurkiransinghk@gmail.com>
Signed-off-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Nafees Nazik <84864519+G3root@users.noreply.github.com>
Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>
Co-authored-by: Joe Au-Yeung <65426560+joeauyeung@users.noreply.github.com>
Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>
Co-authored-by: Bailey Pumfleet <bailey@pumfleet.co.uk>
Co-authored-by: Pranjal Goyal <pranjalgoyal13@gmail.com>
Co-authored-by: Mendy Landa <54242706+MendyLanda@users.noreply.github.com>
Co-authored-by: Crowdin Bot <support+bot@crowdin.com>
Co-authored-by: Gurkiran Singh <61521805+g4rry420@users.noreply.github.com>
Co-authored-by: Udit Takkar <udit.07814802719@cse.mait.ac.in>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: zomars <zomars@me.com>
Co-authored-by: vaibhav bisht <39329182+vaibhav135@users.noreply.github.com>
Co-authored-by: vaibhav bisht <vaibhav@aus.co.in>
Co-authored-by: Keith Williams <keithwillcode@gmail.com>
Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>
Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: Shivang Sharma <70644519+shivang-sharma@users.noreply.github.com>
Co-authored-by: Santosh Kuchetti <106766496+santosh-kuchetti@users.noreply.github.com>
Co-authored-by: Abdurrahman Rajab <rajab-2@windowslive.com>
Co-authored-by: Mahesh Bansode <47536121+MaheshB77@users.noreply.github.com>
Co-authored-by: alannnc <alannnc@gmail.com>
Co-authored-by: Vedant Jain <vedantjainben10@gmail.com>
Co-authored-by: Chiranjeev Vishnoi <66114276+Chiranjeev-droid@users.noreply.github.com>
Co-authored-by: Leo Giovanetti <hello@leog.me>
Co-authored-by: Jeroen Reumkens <hello@jeroenreumkens.nl>
Co-authored-by: Udit Takkar <53316345+Udit-takkar@users.noreply.github.com>
Co-authored-by: Peer Richelsen <peer@cal.com>

zomars commented:
Closing in favor of #9105
